docs/apache-airflow/upgrading-to-2.rst

```diff
@@ -0,0 +1,1164 @@
+'''''''''''''''''''''''''''''''''''''''''''''''''
+
+After upgrading to Airflow 1.10.14, we recommend that you install the "upgrade check" scripts. These scripts will read through your ``airflow.cfg`` and all of your Dags and will give a detailed report of all changes required before upgrading. We are testing this script diligently, and our goal is that any Airflow setup that can pass these tests will be able to upgrade to 2.0 without any issues.
+
+.. code-block:: bash
+
+     pip install apache-airflow-upgrade-check
+
+Once this is installed, please run the upgrade check script.
+
+.. code-block:: bash
+
+     airflow upgrade_check
+
+More details about this process are here  :ref:`Upgrade Check Scripts<upgrade-check:upgrade_check>`
+
+
+Step 4: Set Operators to Backport Providers
+'''''''''''''''''''''''''''''''''''''''''''
+
+Now that you are set up in Airflow 1.10.14 with Python a 3.6+ environment, you are ready to start porting your DAGs to Airflow 2.0 compliance!
+
+The most important step in this transition is also the easiest step to do in pieces. All Airflow 2.0 operators are backwards compatible with Airflow 1.10
+using the [backport providers](./backport-providers.rst) packages. In your own time, you can transition to using these backport-providers
+by pip installing the provider via ``pypi`` and changing the import path.
+
+For example: While historically you might have imported the DockerOperator in this fashion:
+
+.. code-block:: python
+
+    from airflow.operators.docker_operator import DockerOperator
+
+You would now run this command to install the provider:
+
+.. code-block:: bash
+
+    pip install apache-airflow-backport-providers-docker
+
+
+and then import the operator with this path:
+
+.. code-block:: python
+
+    from airflow.providers.docker.operators.docker import DockerOperator
+
+Pleaes note that the backport provider packages are just backports of the provider packages compatible with Airflow 2.0.
+Those provider packages are installed automatically when you install Airflow with extras.
+For example:
+
+.. code-block:: bash
+
+    pip install 'apache-airflow[docker]'
+
+automatically installs the ``apache-airflow-providers-docker`` package.
+But you can manage/upgrade/remove provider packages separately from the Airflow core.
+
+
+Step 5: Upgrade Airflow DAGs
+''''''''''''''''''''''''''''
+
+**Change to undefined variable handling in templates**
+
+Prior to Airflow 2.0 Jinja Templates would permit the use of undefined variables. They would render as an
+empty string, with no indication to the user an undefined variable was used. With this release, any template
+rendering involving undefined variables will fail the task, as well as displaying an error in the UI when
+rendering.
+
+The behavior can be reverted when instantiating a DAG.
+
+.. code-block:: python
+
+    import jinja2
+
+    dag = DAG('simple_dag', template_undefined=jinja2.Undefined)
+
+
+**Changes to the KubernetesPodOperator**
```

Comment:
Should these be headings so they show up in the toc?

```suggestion
Changes to the KubernetesPodOperator
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
I don't know about this -- it might be a bit too noisy with that.

Reply:
Yes, I was worried it would get way too noisy, with too many sections. 
I went back and forth on this a bit, can easily changed it to be sub-headings, but ended up keeping it like this for now.